apps/web-giddh/src/app/gst/gst-setting/gst-setting.component.ts

```diff
@@ -237,29 +231,71 @@ export class GstSettingComponent implements OnInit, OnDestroy {
-                mappingForm.get('fromDate')?.patchValue(this.fromDate);
-                mappingForm.get('toDate')?.patchValue(this.toDate);
+        if (index === 0 && !mappingForm.get('lutNumber')?.value) {
+            mappingForm.get('lutNumber')?.patchValue(null);
+            mappingForm.get('fromDate')?.patchValue(this.fromDate);
+            mappingForm.get('toDate')?.patchValue(this.toDate);
+            if (this.responseArray?.length) {
                 this.responseArray[index]['message'] = null;
             }
+            return;
         } else {
-            if (mappingForm.get('uniqueName')?.value) {
-                this.responseArray[index] = [];
-                this.componentStore.deleteLutNumber({ lutNumberUniqueName: mappingForm.get('uniqueName')?.value });
-                mappings.removeAt(index);
-            }
+            const dialogRef = this.dialog.open(ConfirmModalComponent, {
+                width: '540px',
+                data: {
+                    title: this.commonLocaleData?.app_confirmation,
+                    body: this.localeData?.confirm_delete_message,
+                    ok: this.commonLocaleData?.app_yes,
+                    cancel: this.commonLocaleData?.app_no
+                }
+            });
+            dialogRef.afterClosed().pipe(take(1)).subscribe(response => {
+                if (response) {
+                    if (index === 0) {
+                        if (mappingForm.get('uniqueName')?.value) {
+                            this.componentStore.deleteLutNumber({ lutNumberUniqueName: mappingForm.get('uniqueName')?.value });
+                            mappings.removeAt(index);
+                            mappings.push(this.initLutForm({
+                                fromDate: this.fromDate,
+                                toDate: this.toDate
+                            }));
+                            if (this.responseArray?.length) {
+                                this.responseArray[index]['message'] = null;
+                            }
+                        } else {
+                            mappingForm.get('lutNumber')?.patchValue(null);
+                            mappingForm.get('fromDate')?.patchValue(this.fromDate);
+                            mappingForm.get('toDate')?.patchValue(this.toDate);
+                            if (this.responseArray?.length) {
+                                this.responseArray[index]['message'] = null;
+                            }
+                        }
+                    } else {
+                        if (mappingForm.get('uniqueName')?.value) {
+                            if (this.responseArray?.length) {
+                                this.responseArray[index] = [];
+                            }
+                            this.componentStore.deleteLutNumber({ lutNumberUniqueName: mappingForm.get('uniqueName')?.value });
+                            mappings.removeAt(index);
+                        } else {
```

Comment:
if (mappingForm.get('uniqueName')?.value) {
                            if (this.responseArray?.length) {
                                this.responseArray[index] = [];
                            }
                            this.componentStore.deleteLutNumber({ lutNumberUniqueName: mappingForm.get('uniqueName')?.value });  
                        } 
                            mappings.removeAt(index);

Reply:
Keep this line "mappings.removeAt(index);"

out side the if condition and no need to write else block